packages/react-router-dev/vite/plugin.ts

```diff
@@ -738,6 +738,19 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = () => {
         viteConfigEnv = _viteConfigEnv;
         viteCommand = viteConfigEnv.command;
 
+        // a compatibility layer from Vite v6+ and below because
+        // Vite v6 overrides the default resolve.conditions, so we have to import them
+        // and if the export doesn't exist, it means that we're in Vite v5, so an empty array should be used
+        // https://vite.dev/guide/migration.html#default-value-for-resolve-conditions
+        let viteClientConditions: string[] =
+          "defaultClientConditions" in vite
+            ? Array.from(vite.defaultClientConditions)
```

Comment:
Why is `Array.from` used here?

Reply:
To satisfy TypeScript, I can also do `[...vite.defaultClientConditions]`

Reply:
Could do `...vite.defaultClientConditions ?? []` (and similar in all other cases) to inline it?

Reply:
I've done spreading instead, shorter.

Reply:
> Could do `...vite.defaultClientConditions ?? []` (and similar in all other cases) to inline it?

Ah, sorry, I didn't see this. I pushed the change now, let me know if this is what you meant.